tv_apps/tv_shared/components/smart-dialog/script.js

```diff
@@ -0,0 +1,168 @@
+'use strict';
+
+window.SmartDialog = (function(win) {
+
+  // Extend from the HTMLElement prototype
+  var proto = Object.create(HTMLElement.prototype);
+
+  // Bezier curve function
+  var Bezier = function(bezierParam) {
```

Comment:
I'd prefer using verb for function naming unless it's an object that need to be created by `new`.